src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_goes_through_all_retries.cs

```diff
@@ -0,0 +1,176 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_message_goes_through_all_retries : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_not_send_outgoing_messages_using_batched_dispatch()
+        {
+            await Scenario.Define<Context>(c => c.Id = Guid.NewGuid())
+                .WithEndpoint<Endpoint>(b => b.DoNotFailOnErrorMessages()
+                    .When((session, context) => session.SendLocal(new IncomingMessage
+                    {
+                        Id = context.Id,
+                        UseImmediateDispatch = false
+                    }))
+                )
+                .WithEndpoint<ErrorSpy>()
+                .Done(c => c.MessageMovedToErrorQueue)
+                .Repeat(r => r.For<AllNativeMultiQueueTransactionTransports>())
+                .Should(c => Assert.IsFalse(c.OutgoingMessageReceived, "Outgoing message was unexpectedly dispatched"))
+                .Should(c => Assert.AreEqual(4, c.NumberOfProcessingAttempts))
+                .Run();
+        }
+
+        [Test]
+        public async Task Should_not_send_outgoing_messages_using_immediate_dispatch()
```

Comment:
sends using immediate dispatch should not enlist in an existing transaction. These assertions look wrong to me.